src/Lookup/InProcessCachingDataTypeLookup.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * PropertyDataTypeLookup that provides in-process caching.
+ *
+ * @since 3.1
+ *
+ * @licence GNU GPL v2+
+ * @author Katie Filbert < aude.wiki@gmail.com >
+ */
+class InProcessCachingDataTypeLookup implements PropertyDataTypeLookup {
```

Comment:
For an `CachingPropertyDataTypeLookup`, I'd expect it to not contain a specific caching strategy, and instead utilize a caching service.

We are using `InMemoryFoobarLookup` for this elsewhere. I think this name is slightly better than `InProcess`.